commands/user_e2e_test.go

```diff
@@ -136,3 +140,182 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
+		}()
+
+		cmd := &cobra.Command{}
+
+		// create temp file to replace stdin
+		content := []byte("YES\nYES\n")
+		tmpfile, err := ioutil.TempFile("", "inputfile")
+		s.Require().Nil(err)
+		defer os.Remove(tmpfile.Name()) // remove temp file
+
+		_, err = tmpfile.Write(content)
+		s.Require().Nil(err)
+		_, err = tmpfile.Seek(0, 0)
+		s.Require().Nil(err)
+
+		// replace stdin to do input in testing
+		oldStdin := os.Stdin
+		defer func() { os.Stdin = oldStdin }() // restore
+		os.Stdin = tmpfile
+
+		newUser := s.th.CreateUser()
+		err = deleteUsersCmdF(c, cmd, []string{newUser.Email})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 1)
+		s.Len(printer.GetErrorLines(), 0)
+
+		deletedUser := printer.GetLines()[0].(*model.User)
+		s.Require().Equal(newUser.Username, deletedUser.Username)
+
+		// expect user deleted
+		_, err = s.th.App.GetUser(newUser.Id)
+		s.Require().NotNil(err)
+		s.Require().Equal(err.Error(), "SqlUserStore.Get: Unable to find the user., user_id=store.sql_user.missing_account.const, sql: no rows in result set")
+	})
+
+	s.RunForSystemAdminAndLocal("Delete nonexistent user", func(c client.Client) {
+		printer.Clean()
+		emailArg := "nonexistentUser@example.com"
+
+		previousVal := s.th.App.Config().ServiceSettings.EnableAPIUserDeletion
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableAPIUserDeletion = true })
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableAPIUserDeletion = *previousVal })
+		}()
+
+		cmd := &cobra.Command{}
+		confirm := true
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		err := deleteUsersCmdF(s.th.Client, cmd, []string{emailArg})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Equal("Unable to find user '"+emailArg+"'", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Delete user without permission", func() {
+		printer.Clean()
+
+		previousVal := s.th.App.Config().ServiceSettings.EnableAPIUserDeletion
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableAPIUserDeletion = true })
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableAPIUserDeletion = *previousVal })
+		}()
+
+		cmd := &cobra.Command{}
+		confirm := true
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		newUser := s.th.CreateUser()
+		err := deleteUsersCmdF(s.th.Client, cmd, []string{newUser.Email})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], fmt.Sprintf("Unable to delete user '%s' error: : You do not have the appropriate permissions., ", newUser.Username))
+
+		// expect user not deleted
+		user, err := s.th.App.GetUser(newUser.Id)
+		s.Require().Nil(err)
+		s.Require().Equal(newUser.Username, user.Username)
+	})
+
+	s.Run("Delete user with disabled config", func() {
```

Comment:
I think we should rename this test to "Delete user with disabled config as system admin client"

Reply:
sure, it's more clear that way